core/src/test/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultDriverConfigLoaderTest.java

```diff
@@ -208,4 +208,23 @@ public void should_load_from_file() {
     assertThat(config.getString(DefaultDriverOption.REQUEST_SERIAL_CONSISTENCY))
         .isEqualTo(DefaultConsistencyLevel.SERIAL.name());
   }
+
+  @Test
+  public void should_return_failed_future_if_reloading_not_supported() {
+    DefaultDriverConfigLoader loader =
+        new DefaultDriverConfigLoader(() -> ConfigFactory.parseString(configSource.get())) {
+          @Override
+          public boolean supportsReloading() {
+            return false;
+          }
+        };
+    CompletionStage<Boolean> stage = loader.reload();
+    assertThatStage(stage)
+        .isFailed(
```

Comment:
isn't this breaking backward compatibility? 
If the `reload()` method is called anywhere in our driver code a client did override the `supportsReloading` to return `false` before, he will get the `UnsupportedOperationException` in their code (whereas it was not a case before this change)

Reply:
Yes but if they did override `supportsReloading`, they should expect `reload` to _not_ reload anything and return a failed future, which is not the case currently.